app/cdap/components/GridTable/index.tsx

```diff
@@ -233,10 +262,58 @@ export default function GridTable() {
     setRowsDataList(rowData);
   };
 
+  const isParsingPanel =
+    dataprep?.insights?.name &&
+    isFirstWrangle &&
+    connectorType === 'File' &&
+    Array.isArray(gridData?.headers) &&
+    gridData?.headers.length !== 0;
+
   useEffect(() => {
     getGridTableData();
   }, [gridData]);
 
+  const addDirectives = (directive: string) => {
+    setLoading(true);
+    if (directive) {
```

Comment:
why wouldn't there be a directive?

Reply:
if in case we need to handle error on front end without any API call this will be useful